pkg/cloudclient/aws/aws.go

```diff
@@ -0,0 +1,212 @@
+	// is used to create other clients in order to perform the necessary role chaining to
+	// access a customer's account via STS assume roles.
+	awsClient awsclient.AwsClient
+
+	// ocmClient is a client that can query OCM in order to get the AccountClaim
+	ocmClient ocm.Client
+
+	// awsClientBuilder is responsible for building the aws client in order to
+	// allow mocking during testing
+	awsClientBuilder awsclient.BuilderIface
+
+	// region is the region in which we are building sts clients
+	region string
+}
+
+// NewCloudClient makes a new AWS CloudClient based on the ClusterDeployment and
+// handles initial configuration building for a single request.
+func NewCloudClient(cloudConfig *interface{}, ocmClient ocm.Client, cd *hivev1.ClusterDeployment) (*CloudClient, error) {
+	region := defaultRegion
+	if cd.Spec.Platform.AWS != nil {
+		// overwrite default region if it's set
+		glog.V(8).Infof("Overwriting default region with %s", cd.Spec.Platform.AWS.Region)
+		region = cd.Spec.Platform.AWS.Region
+	}
+
+	accessKeyID, secretAccessKey, err := getAWSCredentialsFromDirectory(credentialDir)
+	if err != nil {
+		glog.Errorf("Couldn't obtain credentials from secret. Error: %s", err.Error())
+		return nil, &ccerrors.InternalError{Err: err}
+	}
+
+	awsClient, err := singletonClient.getAwsClientBuilder().New(accessKeyID, secretAccessKey, "", region)
+	if err != nil {
+		glog.Errorf("Couldn't create AWS cloud client: %s", err.Error())
+		return nil, &ccerrors.UpstreamError{Err: err}
+	}
+
+	return &CloudClient{
+		clusterDeployment: cd,
+		awsClient:         awsClient,
+		ocmClient:         ocmClient,
+		region:            region,
+	}, nil
+}
+
+// The singleton client defined here is used to build the initial cloud client for a request.
+// the biggest use for this client is to be able to inject the awsclient builder during unit testing
+var singletonClient = &CloudClient{}
+
+func (c *CloudClient) getAwsClientBuilder() awsclient.BuilderIface {
+	if c.awsClientBuilder != nil {
+		return c.awsClientBuilder
+	}
+	return &awsclient.Builder{}
+}
+
+func getAWSCredentialsFromDirectory(dir string) (string, string, error) {
+	accessKeyIDFilename := "aws_access_key_id"
+	secretAccessKeyIDFilename := "aws_secret_access_key"
+
+	// ensure that the directory has no trailing slash
+	filepath := strings.TrimSuffix(dir, "/")
+
+	accessKeyBytes, err := ioutil.ReadFile(path.Join(filepath, accessKeyIDFilename))
+	if err != nil {
+		glog.Error(err)
+		return "", "", fmt.Errorf("%s cannot be read from %s", accessKeyIDFilename, filepath)
+	}
+	secretKeyBytes, err := ioutil.ReadFile(path.Join(filepath, secretAccessKeyIDFilename))
+	if err != nil {
+		glog.Error(err)
+		return "", "", fmt.Errorf("%s cannot be read from %s", secretAccessKeyIDFilename, filepath)
+	}
+	accessKeyID := strings.TrimRight(string(accessKeyBytes), "\n")
+	secretKeyID := strings.TrimRight(string(secretKeyBytes), "\n")
+	return accessKeyID, secretKeyID, nil
```

Comment:
Wrap these into a struct so they are more meaningful?
Also makes the return value clearer and reduces flipping them around